src/output/render-manager.cpp

```diff
@@ -849,18 +850,26 @@ class wf::render_manager::impl
 
         for (auto& ds : wf::reverse(repaint.to_render))
         {
+            wlr_surface *sampled_surf = nullptr;
             if (ds->view)
             {
                 repaint.fb.geometry.x = ds->pos.x;
                 repaint.fb.geometry.y = ds->pos.y;
                 ds->view->render_transformed(repaint.fb, ds->damage);
+                sampled_surf = ds->view->get_wlr_surface();
```

Comment:
In this case, all subsurfaces are rendered as well (`view->enumerate_surfaces()` iirc)